pkg/project/controller/controller_test.go

```diff
@@ -38,35 +38,63 @@ func TestSyncNamespaceThatIsTerminating(t *testing.T) {
 		t.Errorf("Unexpected error when handling namespace %v", err)
 	}
 
-	// TODO: we will expect a finalize namespace call after rebase
+	namespaceFinalize := core.NewCreateAction(unversioned.GroupVersionResource{Group: "", Version: "", Resource: "namespaces"}, "", testNamespace)
+	namespaceFinalize.Subresource = "finalize"
+
 	expectedActionSet := []core.Action{
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "buildconfigs"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "policies"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "imagestreams"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "policybindings"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "rolebindings"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "roles"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "routes"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "templates"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "builds"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "namespace"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "deploymentconfig"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "egressnetworkpolicy"}, "", kapi.ListOptions{}),
+		namespaceFinalize,
+		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "", Resource: "buildconfigs"}, testNamespace.Name, kapi.ListOptions{}),
```

Comment:
The fake clients only keep track of resource, they always pass the empty string for group and version.